src/modules/processing/services/tiktok.js

```diff
@@ -20,52 +21,47 @@ export default async function(obj) {
         if (!html) return { error: 'ErrorCouldntFetch' };
 
         if (html.slice(0, 17) === '<a href="https://') {
-            postId = html.split('<a href="https://')[1].split('?')[0].split('/')[3]
-        } else if (html.slice(0, 32) === '<a href="https://m.tiktok.com/v/' && html.includes('/v/')) {
-            postId = html.split('/v/')[1].split('.html')[0].replace("/", '')
+            const fullLink = html.split('<a href="https://')[1].split('?')[0].split('/')
+            username = fullLink[1]
+            postId = fullLink[3]
         }
     }
-    if (!postId) return { error: 'ErrorCantGetID' };
-
-    let deviceInfo = new URLSearchParams(env.tiktokDeviceInfo).toString();
-
-    let apiURL = new URL(apiPath);
-        apiURL.searchParams.append("aweme_id", postId);
+    if (!username || !postId) return { error: 'ErrorCantGetID' };
 
-    let detail = await fetch(`${apiURL.href}&${deviceInfo}`, {
+    // should always be /video/, even for photos
+    const res = await fetch(`${fullDomain}${username}/video/${postId}`, {
         headers: {
-            "user-agent": apiUserAgent
+            "user-agent": genericUserAgent,
+            cookie,
         }
-    }).then(r => r.json()).catch(() => {});
+    })
+    updateCookie(cookie, res.headers)
 
-    detail = detail?.aweme_list?.find(v => v.aweme_id === postId);
-    if (!detail) return { error: 'ErrorCouldntFetch' };
+    const html = await res.text()
+    const json = html
+        .split('<script id="__UNIVERSAL_DATA_FOR_REHYDRATION__" type="application/json">')[1]
+        .split('</script>')[0]
+    const data = JSON.parse(json)
+    const detail = data["__DEFAULT_SCOPE__"]["webapp.video-detail"]["itemInfo"]["itemStruct"]
```

Comment:
probably wrap this in a try-catch and return ErrorCouldntFetch on error since this looks like it would fail pretty commonly